### PR TITLE
Add Tyrants of the Underdark Scoring Strategy

### PR DESCRIPTION
**Pull request by Adelheid483**

### CODE REVIEW

**Comment by Adelheid483** [comment]
Tests runs sucsessfully, but the module doesen't appear

File: server/src/games/Games.TyrantsOfTheUnderdark/TyrantsOfTheUnderdarkScoringModel.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    
+    public byte Deck { get; init; }
+    
+    public byte InnerCircleDeck { get; init; }
+    
+    public byte Tokens { get; init; }
+    
+    public byte ControlSites { get; init; }
+    
+    public byte TotalControlSites { get; init; }
```
Comment (TexnikBy) [inline server/src/games/Games.TyrantsOfTheUnderdark/TyrantsOfTheUnderdarkScoringModel.cs]: Let's sort and name it like a rule book:

- The VP value of each site you control (ControlSites rename to **ControlSitesValue**);
- 2 VP for each site under your total control (TotalControlSites rename to **CountOfTotalControlSites**);
- 1 VP for each troop in your trophy hall (TrophyHall rename to **CountOfTroopInTrophyHall**);
- The deck VP value for each card in your deck, hand, and discard pile (Deck rename to **DeckValue**);
- The inner-circle VP value for each card in your inner circle (InnerCircleDeck rename to **InnerCircleValue**);
- VP tokens gained during the game (Tokens rename to **VictoryPointTokens**).

P.S. don't forget rename [in test](https://github.com/TexnikBy/BGS/pull/20/files#diff-5647a151939579275932fe1e574ff915a0de59bc940d8f1ad4167252040c13dbR33-R38)

File: server/src/games/Games.TyrantsOfTheUnderdark/TyrantsOfTheUnderdarkScoringStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    public int Calculate(JsonElement scoringData)
+    {
+        var model = scoringData.Deserialize<TyrantsOfTheUnderdarkScoringModel>();
+
+        return model.TrophyHall +
+            model.Deck +
+            model.InnerCircleDeck +
+            model.Tokens +
+            model.ControlSites +
+            model.TotalControlSites;
```
Comment (TexnikBy) [inline server/src/games/Games.TyrantsOfTheUnderdark/TyrantsOfTheUnderdarkScoringStrategy.cs]: I suggest the "micro" simplification of scoring on the user side and the general idea of the project to make scoring as simple as possible, make the multiplication here by 2.
After it add tests were:
- multiply by 0;
- multiply by 1;
- multiply by 2.

File: server/tests/games/Games.TyrantsOfTheUnderdark.Tests/Games.TyrantsOfTheUnderdark.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    </ItemGroup>
+
+    <ItemGroup>
+        <ProjectReference Include="..\..\..\src\games\Games.TyrantsOfTheUnderdark\Games.TyrantsOfTheUnderdark.csproj" />
+        <ProjectReference Include="..\..\Tests.Infrastructure\Tests.Infrastructure.csproj" />
+        <ProjectReference Include="..\..\..\src\games\Games.TyrantsOfTheUnderdark\Games.TyrantsOfTheUnderdark.csproj" />
```
Comment (TexnikBy) [inline server/tests/games/Games.TyrantsOfTheUnderdark.Tests/Games.TyrantsOfTheUnderdark.Tests.csproj]: remove duplicate line (Line 15 is the same)

**Review by TexnikBy — DISMISSED** [review]
LGFM
a few comments on renaming to match the rules of the game.

**Review by TexnikBy — APPROVED** [review]

**Pull request merged**